firebase-common/src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -142,6 +145,20 @@ public FirebaseOptions getOptions() {
     return options;
   }
 
+  /**
+   * Returns the specified {@link EmulatorSettings} or a default.
+   *
+   * <p>TODO(samstern): Un-hide this once Firestore, Database, and Functions are implemented
+   *
+   * @hide
+   */
+  @NonNull
+  public EmulatorSettings getEmulatorSettings() {
+    checkNotDeleted();
+    emulatorSettings.freeze();
```

Comment:
If this remains hidden, then the frozen state could just be a simple atomic boolean here. If this is ever public, then this won't work since we wouldn't want to freeze the settings just because this method was called. I would vote we go the simple route now and use a single (atomic) boolean that manages the state in Firebase app.